homeassistant/components/notify/xmpp.py

```diff
@@ -93,29 +118,217 @@ def __init__(self):
+            if data:
+                # Init XEPs for image sending
+                self.register_plugin('xep_0030')  # OOB dep
+                self.register_plugin('xep_0066')  # Out of Band Data
+                self.register_plugin('xep_0071')  # XHTML IM
+                self.register_plugin('xep_0128')  # Service Discovery
+                self.register_plugin('xep_0363')  # HTTP upload
 
             self.connect(force_starttls=self.force_starttls, use_ssl=False)
 
-        def start(self, event):
+        async def start(self, event):
             """Start the communication and sends the message."""
-            self.get_roster()
-            self.send_presence()
+            # Sending image and message independently from each other
+            if data:
+                await self.send_file(timeout=timeout)
+            if message:
+                self.send_text_message()
+
+            self.disconnect(wait=True)
+
+        async def send_file(self, timeout=None):
+            """Send file via XMPP.
+
+            Send XMPP file message using OOB (XEP_0066) and
+            HTTP Upload (XEP_0363)
+            """
             if room:
-                _LOGGER.debug("Joining room %s", room)
                 self.plugin['xep_0045'].join_muc(room, sender, wait=True)
-                self.send_message(mto=room, mbody=message, mtype='groupchat')
+
+            try:
+                # Uploading with XEP_0363
+                _LOGGER.debug("Timeout set to %ss", timeout)
+                url = await self.upload_file(timeout=timeout)
+                if url is None:
+                    raise FileUploadError("Could not upload file")
+            except (IqError, IqTimeout, XMPPError) as ex:
+                _LOGGER.error("Upload error, could not send message %s", ex)
+            except NotConnectedError as ex:
+                _LOGGER.error("Connection error %s", ex)
+            except FileTooBig as ex:
+                _LOGGER.error(
+                    "File too big for server, could not upload file %s", ex)
+            except UploadServiceNotFound as ex:
+                _LOGGER.error("UploadServiceNotFound: "
+                              " could not upload file %s", ex)
+            except FileUploadError as ex:
+                _LOGGER.error("FileUploadError, could not upload file %s", ex)
+            except requests.exceptions.SSLError as ex:
+                _LOGGER.error("Cannot establish SSL connection %s", ex)
+            except requests.exceptions.ConnectionError as ex:
+                _LOGGER.error("Cannot connect to server %s", ex)
+            except (FileNotFoundError,
+                    PermissionError,
+                    IsADirectoryError,
+                    TimeoutError) as ex:
+                _LOGGER.error("Error reading file %s", ex)
+            except FutTimeoutError as ex:
+                _LOGGER.error("The server did not respond in time, %s", ex)
             else:
```

Comment:
This seems a bit odd, could the `else` part just be moved inside the `try` block to keep things together?